cli/src/main/java/com/devonfw/tools/ide/process/ProcessContextImpl.java

```diff
@@ -313,22 +316,34 @@ private String addExecutable(String exec, List<String> args) {
   private void performLogOnError(ProcessResult result, int exitCode, String interpreter) {
 
     if (!result.isSuccessful() && (this.errorHandling != ProcessErrorHandling.NONE)) {
-      String message = createCommandMessage(interpreter, " failed with exit code " + exitCode + "!");
+      IdeSubLogger subLogger = this.context.error();
+      IdeLogLevel ideLogLevel;
+      String message = createCommandMessage(interpreter, "\nfailed with exit code " + exitCode + "!");
+
       if (this.errorHandling == ProcessErrorHandling.THROW_CLI) {
+        subLogger.log(message);
+        result.log(IdeLogLevel.ERROR, context);
         throw new CliProcessException(message, result);
       } else if (this.errorHandling == ProcessErrorHandling.THROW_ERR) {
+        subLogger.log(message);
+        result.log(IdeLogLevel.ERROR, context);
         throw new IllegalStateException(message);
       }
-      IdeSubLogger level;
+
       if (this.errorHandling == ProcessErrorHandling.LOG_ERROR) {
-        level = this.context.error();
+        ideLogLevel = IdeLogLevel.ERROR;
+        subLogger = this.context.error();
       } else if (this.errorHandling == ProcessErrorHandling.LOG_WARNING) {
-        level = this.context.warning();
+        ideLogLevel = IdeLogLevel.WARNING;
+        subLogger = this.context.warning();
       } else {
-        level = this.context.error();
+        ideLogLevel = IdeLogLevel.ERROR;
+        subLogger = this.context.error();
         this.context.error("Internal error: Undefined error handling {}", this.errorHandling);
       }
-      level.log(message);
+
+      subLogger.log(message);
```

Comment:
why first logging on the `subLogger` and then again via `result.log`?
I like that we introduce this `result.log` method to centralize the logging of `ProcessOutput`.
However, this can result in duplicated log output and if we have many lines or even pages of output, we do not want to get it twice.

Reply:
Removed the subLogger entirely.

Reply:
I've used the subLogger only for the message of the exitcode the result.log logs err and out only. Need to re-add it now.